test/task.h

```diff
@@ -111,9 +111,11 @@ typedef enum {
   }                                                       \
  } while (0)
 
-#define ASSERT_BASE(expr, a, operator, b, type, conv)        \
+#define ASSERT_BASE(a, operator, b, type, conv)              \
  do {                                                        \
-  if (!(expr)) {                                             \
+  type eval_a = (type) a;                                    \
+  type eval_b = (type) b;                                    \
```

Comment:
```suggestion
  type eval_a = (type) (a);                                  \
  type eval_b = (type) (b);                                  \
```
Casts have higher priority than e.g. binary operators. Without the parentheses, the left-hand side gets cast but not the right-hand side.